packages/protocol-parser/src/decodeKey.ts

```diff
@@ -0,0 +1,15 @@
+import { Hex } from "viem";
+import { SchemaToPrimitives, KeySchema } from "./common";
+import { decodeKeyTuple } from "./decodeKeyTuple";
+
+export function decodeKey<TSchema extends KeySchema>(
+  keySchema: TSchema,
+  data: readonly Hex[]
+): SchemaToPrimitives<TSchema> {
+  // TODO: refactor and move all decodeKeyTuple logic into this method so we can delete decodeKeyTuple
```

Comment:
is `decodeKeyTuple` still used somewhere else? If not, should we just move the logic in there in this PR?

Reply:
it's used by `blockLogsToStorage` which is getting refactored in the other PR

didn't wanna refactor everything quite yet to avoid a big rebase of the other PR, just wanted to add new methods and deprecate old ones for an easy first pass

Reply:
updated the original issue to remind me to refactor this: https://github.com/latticexyz/mud/issues/1296